crosscluster/logical: check for empty DLQ and fingerprint in ldr roachtests

msbutler opened the pull request:
Epic: none

Release note: none

msbutler committed:
crosscluster/logical: add CheckEmptyDLQs test util

Epic: none

Release note: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/130718)
<!-- Reviewable:end -->

msbutler committed:
roachtest: fingerprint at end of LDR roachtest

This patch adds a check that the dlq table is empty at the end of the roachtest
and fingerprints the replicating tables.

Epic: none

Release note: none

msbutler commented:
extended CI failed unrelated flake: fingerprint mismatch in `TestRandomTables`. This patch doesn't modify underlying LDR code or add fingerprinting to that unit test, so i think this patch is fine to merge.

azhu-crl reviewed (approved):
LGTM!

msbutler commented:
TFTR!

bors r=azhu-crl
